### PR TITLE
Fixing small issue with reference assemblies not setting BaseIntermediateOutputPath

### PR DESCRIPTION
**Pull request by joperezr**
cc: @ViktorHofer @ericstj 

Eric and I found that there was a small bug causing restore re-running on incremental builds due to the project.assets.json file for the src\*.csproj having the same path as the ref\*.csproj. This meant that when you restored both the ref and src projects during a build, if you try re-running restore it wouldn't be a no-op since the assets file will be regenerated. This small change will ensure that ref\*.csproj assets.json files are unique.

### CODE REVIEW

**Comment by joperezr** [comment]
Another potential optimization that I didn't include in this change but could add is to condition the restore of the tests projects only when `'$(BuildTests)' == 'true'`. That would go on line 25 of the following:

https://github.com/dotnet/corefx/blob/1b16f0f02f423e38b04a895677fcc46f0b91bdcf/build.proj#L20-L30

File: eng/ReferenceAssemblies.props [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
   <PropertyGroup Condition="'$(IsReferenceAssembly)'=='true'">
     <OutputPath>$([MSBuild]::NormalizeDirectory('$(ReferenceAssemblyOutputPath)', '$(MSBuildProjectName)', '$(Configuration)'))</OutputPath>
+    <BaseIntermediateOutputPath>$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)', 'ref', '$(MSBuildProjectName)'))</BaseIntermediateOutputPath>
     <IntermediateOutputPath>$([MSBuild]::NormalizeDirectory('$(ArtifactsObjDir)', 'ref', '$(MSBuildProjectName)', '$(Configuration)'))</IntermediateOutputPath>
```
Comment (ericstj) [inline eng/ReferenceAssemblies.props]: Do you know if this lets you remove IntermediateOutputPath and let that be derived from BaseIntermediateOutputPath?  Run a PP and see if the ordering allows.

**Comment by ViktorHofer** [comment]
> Another potential optimization that I didn't include in this change but could add is to condition the restore of the tests projects only when '$(BuildTests)' == 'true'. That would go on line 25 of the following:

That wouldn't work when you first restore and build with: `.\build.cmd -restore -build` (or just .\build.cmd) and later in a subsequent command do a `.\build.cmd -buildtests` because we currently don't trigger the implicit restore when building the test projects: https://github.com/dotnet/corefx/blob/master/src/tests.proj#L34

I believe the restore switch should not be conditioned as it should restore everything upfront.

File: eng/ReferenceAssemblies.props (hunk 1 above)
Comment (ViktorHofer) [reply eng/ReferenceAssemblies.props]: I just checked and we still need to set both as the changes in ProjectLayout.props in Arcade happen earlier: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/ProjectLayout.props#L17-L19

**Comment by ViktorHofer** [comment]
@joperezr I added the BaseOutputPath to mimic what Arcade is doing and applied a few formatting fixes.

**Pull request merged**